<launch>

<arg name = 'cmd_src' default = 'none' />

<param name= "rate" type="double" value="500"/>
<param name="wheel_left" type="string" value="red_wheel_left_joint" />
<param name="wheel_right" type="string" value="red_wheel_right_joint" />
<param name="body_id" type="string" value="red_base_footprint" />
<param name="odom_id" type="string" value="odom" />
<rosparam command="load" file="$(find nusim)/config/initial.yaml" />
<rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />

<include file="$(find nuturtle_description)/launch/load.launch">
    <arg name="color" value="red"/>
    <arg name = 'use_rviz' value = 'false' />
    <arg name = 'use_jsp' value = 'false' />
</include>

<include file="$(find nuturtle_description)/launch/load.launch">
    <arg name="color" value="blue"/>
    <arg name = 'use_rviz' value = 'false' />
    <arg name = 'use_jsp' value = 'true' />
</include>

<node name="nusim_node" pkg="nusim" type="nusim_node" output = "screen"> 
    <rosparam command="load" file="$(find nusim)/config/initial.yaml" />
</node>

<node name="turtle_interface" pkg="nuturtle_control" type="turtle_interface" output = "screen"> 
</node>

<node name="odometry" pkg="nuturtle_control" type="odometry" output = "screen"> 
    <rosparam command="load" file="$(find nusim)/config/initial.yaml" />
</node>

<group if="$(eval cmd_src == 'circle')">
    <node name="circle" pkg="nuturtle_control" type="circle" output = "screen" /> 
</group>

<node pkg="tf2_ros" type="static_transform_publisher" name="world_to_odom" args="0 0 0 0 0 0 world odom" />

<node name="rviz" pkg ="rviz" type="rviz" args="-d $(find nusim)/config/nusim.rviz" />

</launch>